### ISSUE TITLE
About knowledge distillation

### ISSUE THREAD

**Issue by XU-GITHUB-curry** [suous/RepNeXt#1]
Great job!

And I want to know the Section 4.1  _**Results with knowledge distillation**_ and _**Results without knowledge distillation**_ implementation in this paper? How to implement knowledge distillation and what teacher network to use？

**Comment by suous** [comment]
Thanks, we followed previous works: `EfficientFormerV2,` `RepViT,` `SwiftFormer,` and used `RegNetY-16GF` model with a top-1 accuracy of 82.9% as the teacher model for hard distillation.

```python
outputs, outputs_kd = model(inputs)
with torch.no_grad():
    teacher_outputs = teacher_model(inputs)
base_loss = base_criterion(outputs, labels)
distillation_loss = cross_entropy(outputs_kd, teacher_outputs.argmax(dim=1))
loss = base_loss * (1 - alpha) + distillation_loss * alpha
```

**Comment by XU-GITHUB-curry** [comment]
I appreciate your kind reply
&nbsp;


Dr&nbsp;Sco.
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年7月24日(星期三) 晚上7:33
收件人: ***@***.***&gt;; 
抄送: "Dr ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [suous/RepNeXt] About knowledge distillation (Issue #1)














 
Thanks, we followed previous works: EfficientFormerV2, RepViT, SwiftFormer, and used RegNetY-16GF model with a top-1 accuracy of 82.9% as the teacher model for hard distillation.
 outputs, outputs_kd = model(inputs) with torch.no_grad():     teacher_outputs = teacher_model(inputs) base_loss = base_criterion(outputs, labels) distillation_loss = cross_entropy(outputs_kd, teacher_outputs.argmax(dim=1)) loss = base_loss * (1 - alpha) + distillation_loss * alpha

**Comment by chairman-lu** [comment]
Hello, I wonder if the value of --distillation-type is hard and the value of --distillation-alpha is 0.5?

**Comment by suous** [comment]
> Hello, I wonder if the value of --distillation-type is hard and the value of --distillation-alpha is 0.5?

Yes.

**Comment by chairman-lu** [comment]
Thank you
